src/polhemus_ros_driver/knuckle_position_calibration_action_server.py

```diff
@@ -422,27 +496,50 @@ def _calibration(self, goal: CalibrateGoal):
         self._reset_data(hand)
         self._remove_all_markers(hand)
         rospy.loginfo("Starting calibration..")
+        # Clear saved tf messages from previous calibration
+        self._saved_tf_msgs = []
 
         start = rospy.Time.now().to_sec()
         _feedback = CalibrateFeedback()
         _result = CalibrateResult()
 
-        sub = rospy.Subscriber("/tf", TFMessage, self._load_tf_callback, queue_size=10)
+        if self._testing_bag_file_path == '':
```

Comment:
As previously mentioned, would replace this by 
```suggestion
        if self._testing_bag_file_path is None:
```